Remove unneeded example modules

jidicula opened the pull request:
## Checklist

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied the relevant labels to this PR
- [ ] I've added relevant tests for my contribution
- [x] I've updated the documentation and/or added correct docstrings
- [x] I've assigned a reviewer

<!--- For the title, please observe the following rules:
	- Provide a concise and self-descriptive title
	- Do not include the applicable issue number in the title, do it in the PR body
	- If the PR is not ready for review, convert it to a draft.
-->

## Description
<!-- describe what the PR is about. Explain the approach and possible drawbacks.It's ok to repeat some text from the related issue. -->
**Why this change was necessary**
The modules `b0map.py` and `referencemaps.py` were examples at the
beginning of the project and are no longer needed.

**What this change does**
Removes the modules `b0map.py` and `referencemaps.py` as well as any
references to them in the docs and `setup.py`.

**Any side-effects?**
None

## Linked issues
<!-- If the PR fixes any issues, indicate it here with issue-closing keywords: e.g. Resolves #XX, Fixes #XX, Addresses #XX. Note that if you want multiple issues to be autoclosed on PR merge, you must use the issue-closing verb before each relevant issue: e.g. Resolves #1, Resolves #2 -->

**Additional context/notes/links**

Resolves #70

jcohenadad commented on +b0shim/+compute/siemens_basis.m:
```diff
@@ -0,0 +1,160 @@
+%
+% INPUTS
+%
+%    orders
```
i would like very much to see the type indicated for each variable. This is super useful for people not familiar with the code. I've suggested it in this PR (https://github.com/shimming-toolbox/shimming-toolbox/pull/145/files#diff-f7c3aae9cf7ffff5e623bbb43d946a50), which we should merge at some point.

rtopfer replied:
since matlab isn't strongly typed, we need a convention to indicate generic variables (e.g. `numeric` to encompass `double`, `single`, `int`, etc.?) 

it's also a bit unfortunate since, for instance, `orders` in this case should be integers — but doesn't need to be *typed* as such, so suggesting the specific type is a bit misleading...

jcohenadad replied:
discussion moved here: https://github.com/shimming-toolbox/shimming-toolbox/pull/145#issuecomment-652653888

jcohenadad commented on +b0shim/+compute/siemens_basis.m:
```diff
@@ -0,0 +1,160 @@
+% distance from iso/origin to adopted reference point [units: mm]
+r = [1 1 1 1 sqrt(2) sqrt(2) 1 sqrt(2)] ;
+
+% invert polarity of certain terms:
```
any refs here would be welcome

rtopfer replied:
Hmm... none available, just worked this out myself^

jcohenadad replied:
in that case i would add something like: "(found empirically)". just for transparency.

jcohenadad commented on +b0shim/+compute/siemens_basis.m:
```diff
@@ -0,0 +1,160 @@
+% invert polarity of certain terms:
+sh(:,:,:,[2,3,5,8]) = -sh(:,:,:,[2,3,5,8] ) ;
+
+orders = [1 1 1 2 2 2 2 2] ;
```
maybe mention in the header that currently the function only works for up to 2nd SH order, and open a feature issue to cover 2+ orders?

jcohenadad commented on +b0shim/+compute/siemens_basis.m:
```diff
@@ -0,0 +1,160 @@
+%
+% 1. For now, `orders` is, in fact, ignored: fixed as [1:2]—which is suitable
+% for the Prisma (presumably other Siemens systems as well)—however, the
+% 3rd-order shims of the Terra should ultimately be accommodated too. (Requires
```
ah! i'm just seeing this-- scrap [this comment](https://github.com/shimming-toolbox/shimming-toolbox/pull/153/files#r448619602) then. 
i would still open an issue-- much more convenient for project management (than having someone open each .m file and look for "todo" fields)

jcohenadad commented on +b0shim/+compute/siemens_basis.m:
```diff
@@ -0,0 +1,160 @@
+%      vector of non-negative integers (`[0:1:n]` yields harmonics up to n-th
+%      order)
+%
+%    X, Y, Z 
```
these are separate input params (not a single array), so i would do:
~~~
%    X (numeric)
%      X-coordinate in the patient-coordinate system (i.e. DICOM reference, units of mm)
%
%    Y (numeric)
%      Y-coordinate in the patient-coordinate system (i.e. DICOM reference, units of mm)
%
%    Z (numeric)
%      Z-coordinate in the patient-coordinate system (i.e. DICOM reference, units of mm)
~~~

jcohenadad replied:
and to suggest that both 2d or 3d work, i would update the usage above:
~~~
%    basis = siemens_basis( orders, X, Y )
%    basis = siemens_basis( orders, X, Y, Z )
~~~

rtopfer replied:
> and to suggest that both 2d or 3d work, i would update the usage above:
> 
> ```
> %    basis = siemens_basis( orders, X, Y )
> %    basis = siemens_basis( orders, X, Y, Z )
> ```

"2d" refers to computing the basis on a single slice—still need the Z coordinates in either case

jcohenadad replied:
> > and to suggest that both 2d or 3d work, i would update the usage above:
> > ```
> > %    basis = siemens_basis( orders, X, Y )
> > %    basis = siemens_basis( orders, X, Y, Z )
> > ```
> 
> "2d" refers to computing the basis on a single slice—still need the Z coordinates in either case

ha! another good example that expliciting what type/size is expected is important for people not familiar with the code (i.e., me 😅).

rtopfer commented:
@jcohenadad on [this note](https://github.com/shimming-toolbox/shimming-toolbox/pull/145#issuecomment-654539575), can we merge?

jidicula committed:
Remove unneeded example modules

**Why this change was necessary**
The modules `b0map.py` and `referencemaps.py` were examples at the
beginning of the project and are no longer needed.

**What this change does**
Removes the modules `b0map.py` and `referencemaps.py` as well as any
references to them in the docs and `setup.py`.

**Any side-effects?**
None

**Additional context/notes/links**

Resolves #70
